### PR TITLE
add lorentz transform to tensor class

### LINKED ISSUE #354: Add lorentz_tranform to all the classes in symbolic module
🐞 **Problem**
Many classes like `MetricTensor`, `RiemannCurvatureTensor` does not have the `lorentz_transform()` function as of now!

🎯 **Goal**
Implement it in a similar way as that in `GenericVector` class in `symbolic/vector.py`

  - [x] `GenericVector`
  - [ ] `MetricTensor`
  - [ ] `ChristoffelSymbols`
  - [ ] `RiemannCurvatureTensor`
  - [ ] `RicciTensor`
  - [ ] `StressEnergyMomentumTensor`
  - [x] `EinsteinTensor`
  - [ ] `WeylTensor`
  - [ ] `SchoutenTensor`

📋  **Steps to solve the problem**

 * Comment below about what you've started working on.
 * Add, commit, push your changes
 * Submit a pull request and add this in comments - `Addresses #<put issue number here>`
 * Ask for a review in comments section of pull request
 * Celebrate your contribution to this project 🎉

### PR DESCRIPTION
**Pull request by michiboo**
Fixes #354

### CODE REVIEW

File: src/einsteinpy/symbolic/christoffel.py [hunk 1]
```diff
@@ -107,3 +107,23 @@ def change_config(self, newconfig="lll", metric=None):
...
+        Performs a Lorentz transform on the tensor.
+
+        Parameters
+        ----------
+            transformation_matrix : ~sympy.tensor.array.dense_ndim_array.ImmutableDenseNDimArray or list
+                Sympy Array or multi-dimensional list containing Sympy Expressions
+
+        Returns
+        -------
+            ~einsteinpy.symbolic.tensor.ChristoffelSymbols
```
Comment (ritzvik) [inline src/einsteinpy/symbolic/christoffel.py]: `~einsteinpy.symbolic.christoffel.ChristoffelSymbols`. Kindly take note that `ChristoffelSymbols` resides in `symbolic/christoffel.py`

File: src/einsteinpy/symbolic/einstein.py [hunk 2]
```diff
@@ -102,6 +102,7 @@ def lorentz_transform(self, transformation_matrix):
 
         Returns
         -------
+
```
Comment (ritzvik) [inline src/einsteinpy/symbolic/einstein.py]: Was this newline required?

File: src/einsteinpy/symbolic/christoffel.py [hunk 3]
```diff
@@ -107,3 +107,23 @@ def change_config(self, newconfig="lll", metric=None):
...
+                lorentz transformed tensor(or vector)
```
Comment (ritzvik) [inline src/einsteinpy/symbolic/christoffel.py]: Remove the `(or vector)`. `ChristoffelSymbols` is guaranteed to be a 3rd order tensor.

File: src/einsteinpy/symbolic/metric.py [hunk 4]
```diff
@@ -108,3 +108,21 @@ def lower_config(self):
...
+        Parameters
+        ----------
+            transformation_matrix : ~sympy.tensor.array.dense_ndim_array.ImmutableDenseNDimArray or list
+                Sympy Array or multi-dimensional list containing Sympy Expressions
+
+        Returns
+        -------
+            ~einsteinpy.symbolic.tensor.MetricTensor
+                lorentz transformed tensor(or vector)
+
```
Comment (ritzvik) [inline src/einsteinpy/symbolic/metric.py]: `~einsteinpy.symbolic.metric.MetricTensor` due to the reasons stated previously.
remove `or vector`

File: src/einsteinpy/symbolic/ricci.py [hunk 5]
```diff
@@ -92,6 +92,26 @@ def from_christoffels(cls, chris, parent_metric=None):
...
+        Parameters
+        ----------
+            transformation_matrix : ~sympy.tensor.array.dense_ndim_array.ImmutableDenseNDimArray or list
+                Sympy Array or multi-dimensional list containing Sympy Expressions
+
+        Returns
+        -------
+            ~einsteinpy.symbolic.tensor.RicciTensor
+                lorentz transformed tensor(or vector)
+
```
Comment (ritzvik) [inline src/einsteinpy/symbolic/ricci.py]: Same problems as discussed :)

File: src/einsteinpy/symbolic/riemann.py [hunk 6]
```diff
@@ -125,3 +125,23 @@ def change_config(self, newconfig="llll", metric=None):
...
+        Parameters
+        ----------
+            transformation_matrix : ~sympy.tensor.array.dense_ndim_array.ImmutableDenseNDimArray or list
+                Sympy Array or multi-dimensional list containing Sympy Expressions
+
+        Returns
+        -------
+            ~einsteinpy.symbolic.tensor.RiemannCurvatureTensor
+                lorentz transformed tensor(or vector)
+
```
Comment (ritzvik) [inline src/einsteinpy/symbolic/riemann.py]: Here also :)

File: src/einsteinpy/symbolic/ricci.py (hunk 5 above)
Comment (ritzvik) [reply src/einsteinpy/symbolic/ricci.py]: Kindly take the same precautions for other modules you have changed.

File: src/einsteinpy/tests/test_symbolic/test_christoffel.py [hunk 7]
```diff
@@ -98,3 +98,32 @@ def test_properties():
...
+        return list2d
+
+    def get_tensor():
+        x, y, z, w = symbols("x y z w")
+        list2d = [[0 for t1 in range(4)] for t2 in range(4)]
+        list2d[0][0], list2d[0][1], list2d[1][0], list2d[1][1] = x, z, z, x
+        list2d[2][2], list2d[3][3] = y, y
+        return ChristoffelSymbols(
+            list2d, syms=(x, y, z, w), config="lu", parent_metric=None
+        )
```
Comment (ritzvik) [inline src/einsteinpy/tests/test_symbolic/test_christoffel.py]: `ChristoffelSymbols` is a third order tensor while you declared a second order tensor. Anyway, I have a better solution to get an instance of this class. There are many predefined metrics and we can use one of them to get an instance of `ChristoffelSymbols`. In the `get_tensor()` function, do this.

```
metric = AntiDeSitter()
return ChristoffelSymbols.from_metric(metric)
```

To get `AntiDeSitter` kindly do `from einsteinpy.symbolic.predefined import AntiDeSitter` .
Let's see what happens

**Comment by ritzvik** [comment]
tensor for `lorentz_transform` is always second order. Anyway, leave the tests. I don't have a proper way to tests lorentz tranforms. I'll open an issue regarding this

**Comment by shreyasbapat** [comment]
@michiboo are you still working on this?

**Comment by shreyasbapat** [comment]
Oh, I got confused, nvm

**Comment by ritzvik** [comment]
@michiboo , thanks for your contribution.
Merging :)

**Pull request merged**